doc/sphinx/language/core/primitive.rst

```diff
@@ -150,7 +150,17 @@ extraction. Instead, it has to be provided by the user (if they want to compile
 or execute the extracted code). For instance, an implementation of this module
 can be taken from the kernel of Coq (see ``kernel/parray.ml``).
 
-Primitive arrays expose a functional interface, but they are internally
-implemented using a persistent data structure :cite:`ConchonFilliatre07wml`.
-Update and access to an element in the most recent copy of an array are
-constant time operations.
+Semantically, the set array operation behaves as if the array was first entirely
+copied, then the copy is modified and finally returned. Operationally however,
+the array copies are avoided by a mechanism loosely related to the decorator
+design pattern. The full details of the implementation are in section 2.3 of
```

Comment:
Fine, we can remove it. It is a pity though, as it's almost the same pattern. And as far as I can tell, the object-oriented patterns are not very well-known in the Coq community. It is an occasion to make a bridge between two communities.

Reply:
If it is clearer we can add a link to the wikipedia page [](https://en.wikipedia.org/wiki/Decorator_pattern)

Reply:
Generally I think it's a good to avoid introducing concepts that are only mentioned in one place in the doc--unless they're really helpful for the explanation or are terms that every reader should know.

I often find design pattern definitions a bit vague and many people only know a few of them.  And they're for object-oriented code--which Coq uses very little (only CoqIDE AFAIK) and so, I would expect, not used here.  (The Wiki page begins with "In object-oriented programming, the decorator pattern is a design pattern that allows behavior to be added to an individual object, dynamically, without affecting the behavior of other objects from the same class.")

What do you think?

Reply:
I'll remove it